client/my-sites/checkout/composite-checkout/wpcom/hooks/use-shopping-cart.ts

```diff
@@ -249,6 +252,24 @@ function shoppingCartHookReducer(
 	return state;
 }
 
+function removeItemFromLocalStorage( productSlug: string ) {
+	const cartItemsFromLocalStorage = JSON.parse(
+		window.localStorage.getItem( 'shoppingCart' ) || '[]'
+	);
+
+	if ( ! isEmpty( cartItemsFromLocalStorage ) ) {
+		const newCart = cartItemsFromLocalStorage.filter(
+			( product ) => product.product_slug !== productSlug
+		);
+
+		try {
+			window.localStorage.setItem( 'shoppingCart', JSON.stringify( newCart ) );
```

Comment:
I think it might be a little misleading to call this variable `newCart` when it's actually an array of products. I mean, I agree that in concept, that is what a cart _is_, but in the context of WP.com, our cart object holds a lot more. What about something like `newCartItems`?

Reply:
Agreed, updated with this change.